### ISSUE TITLE
[Bug] No error raised when using / instead of . for some command-line parameters

### ISSUE THREAD

**Issue by odelalleau** [facebookresearch/hydra#775]
# 🐛 Bug

The following command *seems* to run fine but actually ignores the partition:
```
python my_app.py hydra/launcher=submitit_slurm hydra/launcher/params/partition=dev -m
```

This is because the correct syntax should be `hydra.launcher.params.partition=dev` (no /).

## To reproduce

** Minimal Code/Config snippet to reproduce **

Run https://github.com/facebookresearch/hydra/blob/master/examples/tutorials/structured_configs/1_minimal/my_app.py with the command line above.

** Stack trace/error message **

No error: it launches the job on the cluster, but not on the dev partition.

## Expected Behavior

An error should be raised to tell the user they messed up.

## System information
- **Hydra Version** :  current master (91c9c289a10a7076b388b61bee4cdcfd4a0e2f84)
- **Python version** : 3.7.7
- **Virtual environment type and version** : conda 4.7.10
- **Operating system** : Linux

**Comment by omry** [comment]
There is an error when overriding the value improperly in each one of the sub jobs. 
I did not try with actual slrum but should be the same.
```
$ python example/my_app.py hydra/launcher=submitit_local hydra/launcher/params/partition=dev  -m 2>&1 | pastebinit 
http://paste.ubuntu.com/p/kn5K4Stbzh/
```

**Comment by omry** [comment]
Something is weird here:

This fails on launching each sweep job:
```
python examples/tutorials/basic/your_first_hydra_app/1_simple_cli/my_app.py  hydra/launcher=submitit_local hydra/launcher/params/x=dev -m
```

And this fails immediately:
```
$ python examples/tutorials/basic/your_first_hydra_app/1_simple_cli/my_app.py  hydra/launcher=basic hydra/launcher/params/x=dev -m
[2020-07-15 11:22:30,059][HYDRA] Launching 1 jobs locally
[2020-07-15 11:22:30,059][HYDRA]        #0 : 
Could not override 'hydra/launcher/params/x'. No match in config.
To append to your config use +hydra/launcher/params/x=dev
Key 'hydra/launcher/params/x' is not in struct
        full_key: hydra/launcher/params/x
        reference_type=Optional[Dict[Any, Any]]
        object_type=dict

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

**Comment by omry** [comment]
will dig later. free free to try to understand why this is behaving differently.

**Comment by odelalleau** [comment]
Ah, I forgot to mention that when I tried my original example, since it queued my job in the wrong partition, it got stuck there and I ended up killing it. So I never reached the point where I would see the error, if it is raised only when the job starts.
In that sense it could be related to #776: ideally the error would be raised immediately.

From your tests it sounds like this behavior is launcher-dependent -- I don't feel like I"m familiar enough with the internals of Hydra to dig into it now (and it's not a top priority issue for me).

Thanks!
